src/transformers/utils/quantization_config.py

```diff
@@ -379,7 +379,11 @@ class GPTQConfig(QuantizationConfigMixin):
             The exllama config. You can specify the version of the exllama kernel through the `version` key. Defaults
             to `{"version": 1}` if unset.
         cache_block_outputs (`bool`, *optional*, defaults to `True`):
-                Whether to cache block outputs to reuse as inputs for the succeeding block.
+            Whether to cache block outputs to reuse as inputs for the succeeding block.
+        modules_in_block_to_quantize (`List[List[str]]`, *optional*):
+            List list of module names to quantize in the block specified. This argument is useful to exclude certain linear modules from being quantized.
+            The block to quantize can be specified by setting `block_name_to_quantize`. We will quantize each list sequentially. If not set, we will quantize all linear layers.
+            Example: `inside_layer_modules=[["self_attention.query_key_value"], ["mlp.dense_h_to_4h"]]`
```

Comment:
```suggestion
            Example: `modules_in_block_to_quantize =[["self_attention.query_key_value"], ["mlp.dense_h_to_4h"]]`
```
So we can only quantize a specific block, but in this block we can quantize a list of layers ? If so why not make the block name to quantize a dictionary that take the name of the layer, and a list of modules. Also not sure I get why the modules in block to quantize needs to be a list of list ?

Reply:
This is related on how the quantization is performed. Let's take an example: 
```python
modules_in_block_to_quantize = [
        ["self_attn.k_proj", "self_attn.v_proj", "self_attn.q_proj"],
        ["self_attn.o_proj"],
        [
            "block_sparse_moe.experts.0.w1",
            "block_sparse_moe.experts.1.w1",
            "block_sparse_moe.experts.2.w1",
            "block_sparse_moe.experts.3.w1",
            "block_sparse_moe.experts.4.w1",
            "block_sparse_moe.experts.5.w1",
            "block_sparse_moe.experts.6.w1",
            "block_sparse_moe.experts.7.w1",
            "block_sparse_moe.experts.0.w2",
            "block_sparse_moe.experts.1.w2",
            "block_sparse_moe.experts.2.w2",
            "block_sparse_moe.experts.3.w2",
            "block_sparse_moe.experts.4.w2",
            "block_sparse_moe.experts.5.w2",
            "block_sparse_moe.experts.6.w2",
            "block_sparse_moe.experts.7.w2",
         ],
        [
            "block_sparse_moe.experts.0.w3",
            "block_sparse_moe.experts.1.w3",
            "block_sparse_moe.experts.2.w3",
            "block_sparse_moe.experts.3.w3",
            "block_sparse_moe.experts.4.w3",
            "block_sparse_moe.experts.5.w3",
            "block_sparse_moe.experts.6.w3",
            "block_sparse_moe.experts.7.w3",
        ]
    ]
```

We first quantize `["self_attn.k_proj", "self_attn.v_proj", "self_attn.q_proj"]` at the same time. Then we quantize `["self_attn.o_proj"]` with q,k,v quantized. So the input will be different compared to the original q,k,v. By doing this,  `["self_attn.o_proj"] ` should be better quantized since we are using the real input he will get.
For the `block_sparse_moe.experts` , it makes sense to first quantize w1 and w3 together before w2 since we have in the related module:
```
current_hidden_states = self.act_fn(self.w1(hidden_states)) * self.w3(hidden_states)
current_hidden_states = self.w2(current_hidden_states)
```

As for the dictionary, I prefer to keep it like this since the order is quite important.

Reply:
Great, thanks for the explanation! List of lists makes sense then. For adding to the doc, just the first example e.g. 

```
        ["self_attn.k_proj", "self_attn.v_proj", "self_attn.q_proj"],
        ["self_attn.o_proj"],
```

is enough to covey the importances of the order of operations

Reply:
I will change to that then !